lib/matplotlib/axis.py

```diff
@@ -1223,18 +1221,29 @@ def get_ticklines(self, minor=False):
         return self.get_majorticklines()
 
     def get_majorticklocs(self):
-        "Get the major tick locations in data coordinates as a numpy array"
+        """Get the array of major tick locations in data coordinates."""
         return self.major.locator()
 
     def get_minorticklocs(self):
-        "Get the minor tick locations in data coordinates as a numpy array"
-        return self.minor.locator()
+        """Get the array of minor tick locations in data coordinates."""
+        # Remove minor ticks duplicating major ticks.
+        major_locs = self.major.locator()
+        minor_locs = self.minor.locator()
```

Comment:
aren't we now calling both locators twice?  Seems inefficient to me.

Reply:
OK, actually I'm confused - why are we calling the locator at all here?  Do we not cache the existing tick locations in some way?  Sorry if I'm being dumb.

Reply:
I... don't think so?  Note that the previous implementation also called the locator.
Obviously we could inspect what are the *current* tick positions, but I don't know if there's any invalidation mechanism in place right now (e.g. what happens when you pan/zoom?).
IOW, this uses the same structure as before the PR; having `set_{major,minor}ticklocs` handle cache invalidation could be future work ( :) ).